packages/apex-ls/src/profiling/ProfilingService.ts

```diff
@@ -0,0 +1,502 @@
+}
+
+/**
+ * Sanitize a tag for use in filenames
+ * Removes invalid characters and limits length
+ */
+function sanitizeTag(tag: string): string {
+  // Remove invalid filename characters: < > : " / \ | ? *
+  // Replace spaces and other special chars with underscores
+  // Limit length to 50 characters
+  return tag
+    .replace(/[<>:"/\\|?*\x00-\x1F]/g, '')
+    .replace(/\s+/g, '_')
+    .substring(0, 50)
+    .trim();
+}
+
+/**
+ * Profiling status
+ */
+export interface ProfilingStatus {
+  isProfiling: boolean;
+  type: ProfilingState;
+  available: boolean;
+}
+
+/**
+ * Service for managing Node.js CPU and heap profiling using the inspector API
+ *
+ * This service dynamically loads the Node.js inspector module to avoid bundling
+ * issues in browser/webworker environments. Profiling is only available in
+ * Node.js (desktop) environments.
+ */
+export class ProfilingService {
+  private static instance: ProfilingService | null = null;
+  private inspector: any = null;
+  private session: any = null;
+  private isConnected = false;
+  private currentState: ProfilingState = 'idle';
+  private outputDir: string | null = null;
+  private logger: LoggerInterface | null = null;
+  private cpuProfileSequence = 1; // Sequence counter for CPU profiles
+  private heapProfileSequence = 1; // Sequence counter for heap profiles
+
+  private constructor() {
+    // Private constructor for singleton pattern
+  }
+
+  /**
+   * Get the singleton instance of ProfilingService
+   */
+  public static getInstance(): ProfilingService {
+    if (!ProfilingService.instance) {
+      ProfilingService.instance = new ProfilingService();
+    }
+    return ProfilingService.instance;
+  }
+
+  /**
+   * Initialize the service with logger and output directory
+   */
+  public initialize(logger: LoggerInterface, outputDir: string): void {
+    this.logger = logger;
+    this.outputDir = outputDir;
+  }
+
+  /**
+   * Check if the inspector API is available in the current environment
+   */
+  public isAvailable(): boolean {
+    if (this.inspector !== null) {
+      return true;
+    }
+
+    // Check if we're in a Node.js environment
+    if (typeof process === 'undefined' || !process.versions?.node) {
+      return false;
+    }
+
+    // Try to load inspector module
+    try {
+      // Use require() for dynamic loading (not at module level)
+      this.inspector = require('inspector');
+      return true;
+    } catch (_error) {
+      return false;
+    }
+  }
+
+  /**
+   * Ensure inspector session is connected
+   */
+  private async ensureConnected(): Promise<void> {
+    if (!this.isAvailable()) {
+      throw new Error('Inspector API is not available in this environment');
+    }
+
+    if (this.isConnected && this.session) {
+      return;
+    }
+
+    try {
+      this.session = new this.inspector.Session();
+      this.session.connect();
+      this.isConnected = true;
+      this.logger?.debug('Inspector session connected');
+    } catch (error) {
```

Comment:
The process to catch and throw the same error here with the log, can we just remove the catch block and allow the error to bubble up?